Jenkinsfile

```diff
@@ -138,7 +147,14 @@ pipeline {
                                 }
                                 withMaven(maven: 'M35', jdk: 'jdk8-latest', globalMavenSettingsConfig: 'default-global-settings', mavenSettingsConfig: 'codice-maven-settings', mavenOpts: '${LARGE_MVN_OPTS} ${LINUX_MVN_RANDOM}') {
                                     withCredentials([string(credentialsId: 'sonarqube-token', variable: 'SONAR_TOKEN')]) {
-                                        sh 'mvn -q -B -Dfindbugs.skip=true -Dcheckstyle.skip=true org.jacoco:jacoco-maven-plugin:prepare-agent install sonar:sonar -Dsonar.host.url=https://sonarqube.com -Dsonar.login=$SONAR_TOKEN  -Dsonar.organization=codice -Dsonar.projectKey=ddf -pl !$DOCS,!$ITESTS'
+                                        script {
+                                            // If this build is not a pull request, run sonar scan. otherwise run incremental scan
+                                            if (env.CHANGE_ID == null) {
+                                                sh 'mvn -q -B -Dfindbugs.skip=true -Dcheckstyle.skip=true org.jacoco:jacoco-maven-plugin:prepare-agent install sonar:sonar -Dsonar.host.url=https://sonarqube.com -Dsonar.login=$SONAR_TOKEN  -Dsonar.organization=codice -Dsonar.projectKey=ddf -pl !$DOCS,!$ITESTS'
+                                            } else {
+                                                sh 'mvn -q -B -Dfindbugs.skip=true -Dcheckstyle.skip=true org.jacoco:jacoco-maven-plugin:prepare-agent install sonar:sonar -Dsonar.host.url=https://sonarqube.com -Dsonar.login=$SONAR_TOKEN  -Dsonar.organization=codice -Dsonar.projectKey=ddf -pl !$DOCS,!$ITESTS -Dgib.enabled=true -Dgib.referenceBranch=/refs/remotes/origin/$CHANGE_TARGET'
```

Comment:
❓ Should this show `preview` mode for the PRs? On the incremental we still don't want to push the results to the sonarqube server that should only be done on the respective branches (master, 2.12.x, etc)

Reply:
Have we ever tried running a sonar scan on an incremental build? I'm curious what that will look like on the sonarqube site, flipping back and forth between full and incremental build scans.

Reply:
Updated with preview mode, should I remove the incremental build options? Does preview mode push scan results up to the sonar site @shaundmorris? If not than incremental should be fine in preview mode